src/sentry/web/frontend/demo_start.py

```diff
@@ -0,0 +1,70 @@
+import petname
+
+from django.http import Http404
+from django.conf import settings
+from django.db import transaction
+from django.views.decorators.csrf import csrf_exempt
+from django.template.defaultfilters import slugify
+
+from sentry import roles
+from sentry.demo.data_population import populate_python_project, populate_react_project
+from sentry.models import (
+    User,
+    Organization,
+    OrganizationMember,
+    OrganizationMemberTeam,
+    Project,
+    ProjectKey,
+)
+from sentry.utils import auth
+from sentry.utils.email import create_fake_email
+from sentry.web.frontend.base import BaseView
+
+
+def generate_random_name():
+    return petname.Generate(2, " ", letters=10).title()
+
+
+class DemoStartView(BaseView):
+    @transaction.atomic
+    @csrf_exempt
+    def dispatch(self, request):
+        if not settings.DEMO_MODE:
+            raise Http404
```

Comment:
Instead of doign this here, It dbe better to not even register the url routes.

Reply:
@mattrobenolt That's what I started with but it wasn't working well with tests since the url file is already loaded. Is there a way to re-register the routes after I use `with self.settings` in the test?

Reply:
Benefit of that being we also would be able to skip the import of this entirely. I think we have examples of this based on `settings.DEBUG` in the urls.py somewhere.

Reply:
No, you'd just not use `with settings()`. We'd set it explicitly to True for tests always.

Reply:
@mattrobenolt Hmmm...so how else do I set `DEMO_MODE` and then have it re-register the routes?

Reply:
If I had to guess, I'd put it in here: https://github.com/getsentry/sentry/blob/master/src/sentry/utils/pytest/sentry.py This is where we set settings configuration for tests.

Reply:
@mattrobenolt Oh, are you saying we should have `DEMO_MODE=True` for all tests? I was trying to only enable it for the tests that require it. Cause I definitely need the spec for if `DEMO_MODE=False` that we return a 404 on the route.

Reply:
Yes, you'd set `settings.DEMO_MODE = True` for all tests in the pytest config. It's not valuable to test that the route isn't there on 404. That's excessive.

Reply:
I think it's fine to validate that when `DEMO_MODE=False` the route is not there. That avoids someone exposing the route by accident somehow

Reply:
I mean, then go for it and figure out how to do it in tests. We don't do this kind of trivial testing elsewhere since it's silly to have a test over a basic `if` statement for url routes. We do this elsewhere in a bunch of places wihtout testing.

https://github.com/getsentry/sentry/blob/master/src/sentry/web/urls.py#L55-L68
All of these are behind a conditional: https://github.com/getsentry/sentry/blob/master/src/sentry/web/debug_urls.py

Which is set https://github.com/getsentry/sentry/blob/master/src/sentry/utils/pytest/sentry.py#L90

Reply:
Ok, I have checking `DEMO_MODE` twice so we can avoid adding the route but still have tests pass. This is because we only load the URL file once when we run tests so even if we disable `DEMO_MODE` for the test, the route is still there. Having the check twice I think is OK considering how absolutely critical it is for this route not to be exposed on SaaS.

Reply:
You could also do somethign more heavy handed if you want like:

```python
if settings.DEMO_MODE:
    class DemoStartView(BaseView):
        ...
```

Then it fundamentally the class doesnt' even exist outside demo mode.